rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -72,10 +72,17 @@ import CfgImpl
 class CallableScopeTree extends StandardTree, PreOrderTree, PostOrderTree, Scope::CallableScope {
   override predicate propagatesAbnormal(AstNode child) { none() }
 
+  private int getNumberOfSelfParams() {
+    if this.getParamList().hasSelfParam() then result = 1 else result = 0
+  }
+
   override AstNode getChildNode(int i) {
-    result = this.getParamList().getParam(i)
+    i = 0 and
+    result = this.getParamList().getSelfParam()
+    or
+    result = this.getParamList().getParam(i - this.getNumberOfSelfParams())
```

Comment:
The indices of `getChildNode` need not be consecutive, nor start from 0, so I would just do `result = this.getParamList().getParam(i - 1)`